generate_analysis_graphs.py

```diff
@@ -70,49 +72,103 @@
+    #  - % Relevant Messages, by computing Relevant Messages / Total Messages * 100, to 1 decimal place.
+    for count in engagement_counts.values():
+        count["% Relevant Messages"] = round(count["Relevant Messages"] / count["Total Messages"] * 100, 1)
+
+    # Export the engagement counts to a csv.
+    with open(f"{output_dir}/engagement_counts.csv", "w") as f:
+        headers = ["Episode", "Total Messages", "Relevant Messages", "% Relevant Messages", "Total Participants"]
+        writer = csv.DictWriter(f, fieldnames=headers, lineterminator="\n")
+        writer.writeheader()
+
+        for row in engagement_counts.values():
+            writer.writerow(row)
+            
+    log.info("Graphing the per-episode engagement counts...")
+    # Graph the number of messages in each episode
+    altair.Chart(
+        altair.Data(values=[{"episode": x["Episode"], "count": x["Total Messages"]}
+                            for x in engagement_counts.values() if x["Episode"] != "Total"])
     ).mark_bar().encode(
-        x=altair.X("show:N", title="Show", sort=list(messages_per_show.keys())),
+        x=altair.X("episode:N", title="Episode"),
         y=altair.Y("count:Q", title="Number of Messages")
     ).properties(
-        title="Messages per Show"
-    )
-    chart.save(f"{output_dir}/messages_per_show.html")
-    chart.save(f"{output_dir}/messages_per_show.png", scale_factor=IMG_SCALE_FACTOR)
-
-    # Compute the number of individuals in each show and graph
-    log.info(f"Graphing the number of individuals who responded to each show...")
-    individuals_per_show = OrderedDict()  # Of radio show index to individuals count
-    for plan in PipelineConfiguration.RQA_CODING_PLANS:
-        individuals_per_show[plan.raw_field] = 0
-
-    for ind in individuals:
-        for plan in PipelineConfiguration.RQA_CODING_PLANS:
-            if ind.get(plan.raw_field, "") != "" and ind["consent_withdrawn"] == "false":
-                individuals_per_show[plan.raw_field] += 1
+        title="Messages per Episode"
+    ).save(f"{output_dir}/messages_per_episode.png", scale_factor=IMG_SCALE_FACTOR)
 
-    chart = altair.Chart(
-        altair.Data(values=[{"show": k, "count": v} for k, v in individuals_per_show.items()])
+    # Graph the number of individuals in each episode
+    altair.Chart(
+        altair.Data(values=[{"episode": x["Episode"], "count": x["Total Participants"]}
+                            for x in engagement_counts.values() if x["Episode"] != "Total"])
     ).mark_bar().encode(
-        x=altair.X("show:N", title="Show", sort=list(individuals_per_show.keys())),
+        x=altair.X("episode:N", title="Episode"),
         y=altair.Y("count:Q", title="Number of Individuals")
     ).properties(
-        title="Individuals per Show"
-    )
-    chart.save(f"{output_dir}/individuals_per_show.html")
-    chart.save(f"{output_dir}/individuals_per_show.png", scale_factor=IMG_SCALE_FACTOR)
+        title="Individuals per Episode"
```

Comment:
rename to Participants per Episode to match rda naming.

Reply:
changed, thanks